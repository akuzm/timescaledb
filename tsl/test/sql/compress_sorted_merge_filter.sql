-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.

create table batches(ts timestamp, id int);

select create_hypertable('batches', 'ts');

alter table batches set (timescaledb.compress, timescaledb.compress_segmentby = 'id');

truncate table batches;

insert into batches values ('2022-02-02 00:00:01', 1), ('2022-02-02 00:00:11', 1),
    ('2022-02-02 00:00:02', 2), ('2022-02-02 00:00:12', 2),
    ('2022-02-02 00:00:03', 3), ('2022-02-02 00:00:13', 3);

select compress_chunk(x, true) from show_chunks('batches') x;

analyze batches;

set timescaledb.debug_require_batch_sorted_merge to true;

select ts from batches where ts != '2022-02-02 00:00:02' order by ts;

set timescaledb.enable_decompression_sorted_merge to off;

select ts from batches where ts != '2022-02-02 00:00:02' order by ts;