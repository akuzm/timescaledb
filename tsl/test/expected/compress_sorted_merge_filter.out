-- This file and its contents are licensed under the Timescale License.
-- Please see the included NOTICE for copyright information and
-- LICENSE-TIMESCALE for a copy of the license.
create table batches(ts timestamp, id int);
select create_hypertable('batches', 'ts');
WARNING:  column type "timestamp without time zone" used for "ts" does not follow best practices
NOTICE:  adding not-null constraint to column "ts"
  create_hypertable   
----------------------
 (1,public,batches,t)
(1 row)

alter table batches set (timescaledb.compress, timescaledb.compress_segmentby = 'id');
truncate table batches;
insert into batches values ('2022-02-02 00:00:01', 1), ('2022-02-02 00:00:11', 1),
    ('2022-02-02 00:00:02', 2), ('2022-02-02 00:00:12', 2),
    ('2022-02-02 00:00:03', 3), ('2022-02-02 00:00:13', 3);
select compress_chunk(x, true) from show_chunks('batches') x;
             compress_chunk             
----------------------------------------
 _timescaledb_internal._hyper_1_1_chunk
(1 row)

analyze batches;
set timescaledb.debug_require_batch_sorted_merge to true;
select ts from batches where ts != '2022-02-02 00:00:02' order by ts;
            ts            
--------------------------
 Wed Feb 02 00:00:01 2022
 Wed Feb 02 00:00:03 2022
 Wed Feb 02 00:00:11 2022
 Wed Feb 02 00:00:12 2022
 Wed Feb 02 00:00:13 2022
(5 rows)

set timescaledb.enable_decompression_sorted_merge to off;
select ts from batches where ts != '2022-02-02 00:00:02' order by ts;
ERROR:  debug: batch sorted merge is required but not used
